go/arrow/cdata/cdata_exports.go

```diff
@@ -385,7 +390,13 @@ func exportArray(arr arrow.Array, out *CArrowArray, outSchema *CArrowSchema) {
 		for i := range bufs {
 			buf := bufs[i]
 			if buf == nil || buf.Len() == 0 {
-				buffers[i] = nil
+				if i > 0 || !internal.HasValidityBitmap(arr.DataType().ID(), flatbuf.MetadataVersionV5) {
+					// apache/arrow#33936: export a dummy buffer to be friendly to
+					// implementations that don't import NULL properly
+					buffers[i] = (*C.void)(unsafe.Pointer(&C.kGoCdataZeroRegion))
+				} else {
+					buffers[i] = nil
+				}
```

Comment:
should this be behind some sort of flag so that we can easily switch back to using `nil` when the other implementations catch up? Or is that not worth it?

Reply:
If/when we decide to remove it later, I think we should just remove it from the code entirely, instead of trying to flag it.

Reply:
The C++ side fix isn't even shipping until Arrow 12